### PR TITLE
fix(storybook): apply a webpack tweak for storybook and angular

### LINKED ISSUE #8360: Nx Version 13 not working with Storybook
## Current Behavior
Currently Storybook cannot be correctly run for libraries created with standard tools in NX.
After creating a workspace, adding a library and Storybook to the library, the Storybook itself doesn't work correctly. Please see reproduction steps for details.

## Expected Behavior
Storybook should work and render stories correctly.
This worked for NX 12 and Storybook 6.3.0.

### Quick Reproduction
I have prepared a GitHub repo reproducing the issue.

1. clone https://github.com/KirillMetrik/story-test
2. run `npm install`
3. run `nx run my-lib:storybook`
4. After build finishes, open the story for "Super" component.

Expected render for the story:
```
Hello, I am super!
```

Actual render for the story:
```
export default "
\r\n Hello, I am super!\r\n
";
```

### Long Reproduction (all steps from scratch)
1. create a new workspace: `npx create-nx-workspace --preset=angular`
2.  create a new library `ng g @nrwl/angular:lib my-lib`
3. add Storybook to the newly created library: `nx g @nrwl/angular:storybook-configuration my-lib`
4. modify references to Storybook in package.json in order to work around issue: https://github.com/storybookjs/storybook/issues/16977 :
```
    "@storybook/addon-essentials": "6.5.0-alpha.4",
    "@storybook/angular": "6.5.0-alpha.4",
    "@storybook/builder-webpack5": "6.5.0-alpha.4",
    "@storybook/manager-webpack5": "6.5.0-alpha.4",
```
5. run `npm install`
6. add a simple component to your library, e.g. with the following template:
```
<div>
    Hello, I am super!
</div>
```
7. add "build" target to angular.json in order to work around this issue: https://github.com/nrwl/nx/issues/8199
```
"build": {
                    "builder": "@nrwl/angular:package",
                    "options": {
                        "tsConfig": "libs/my-lib/tsconfig.lib.json",
                        "project": "libs/my-lib/ng-package.json"
                    },
                    "configurations": {
                        "production": {
                            "tsConfig": "libs/my-lib/tsconfig.lib.prod.json"
                        }
                    }
                },
```
8. add "ng-package.json" to `libs/my-lib` folder and fill it with the following content:
```
{ 
    "$schema": "../../node_modules/ng-packagr/ng-package.schema.json", 
    "dest": "../../dist/libs/my-lib", 
    "lib": { 
        "entryFile": "src/index.ts" 
    }, 
    "allowedNonPeerDependencies": [ 
        "." 
    ] 
}
```
9. Add a simple story for your component:
```
import { Meta, Story } from '@storybook/angular';
import { SuperComponent } from './super.component';

export default {
    title: 'Super',
    component: SuperComponent
} as Meta;

export const Super: Story<SuperComponent> = () => ({
    props: {
    }
});
```
10. run `nx run my-lib:storybook`
11. After build finishes, open the story for our component.

Expected render for the story:
```
Hello, I am super!
```

Actual render for the story:
```
export default "
\r\n Hello, I am super!\r\n
";
```

### Environment
```
Node : 14.17.0
  OS   : win32 x64
  npm  : 6.14.13

  nx : 13.4.2
  @nrwl/angular : 13.4.2
  @nrwl/cli : 13.4.2
  @nrwl/cypress : 13.4.2
  @nrwl/devkit : 13.4.2
  @nrwl/eslint-plugin-nx : 13.4.2
  @nrwl/express : undefined
  @nrwl/jest : 13.4.2
  @nrwl/linter : 13.4.2
  @nrwl/nest : undefined
  @nrwl/next : undefined
  @nrwl/node : undefined
  @nrwl/nx-cloud : undefined
  @nrwl/react : undefined
  @nrwl/react-native : undefined
  @nrwl/schematics : undefined
  @nrwl/tao : 13.4.2
  @nrwl/web : undefined
  @nrwl/workspace : 13.4.2
  @nrwl/storybook : 13.4.2
  @nrwl/gatsby : undefined
  typescript : 4.4.4
  rxjs : 7.4.0
  ---------------------------------------
  Community plugins:
         @angular/animations: 13.1.1
         @angular/common: 13.1.1
         @angular/compiler: 13.1.1
         @angular/core: 13.1.1
         @angular/forms: 13.1.1
         @angular/platform-browser: 13.1.1
         @angular/platform-browser-dynamic: 13.1.1
         @angular/router: 13.1.1
         @angular-devkit/build-angular: 13.1.2
         @angular/cli: 13.1.2
         @angular/compiler-cli: 13.1.1
         @angular/language-service: 13.1.1
         @storybook/angular: 6.5.0-alpha.4
```

### PR DESCRIPTION
**Pull request by rfprod**
- [x] remove html raw-loader from the webpack rules array if storybook is used with angular v13;
- [x] update related unit test snapshots; 

the html raw-loader breaks jit compilation when storybook 6.5 aplha is used with angular v13

ISSUES CLOSED: #8360

## Related Issue(s)

Fixes #8360

### CODE REVIEW

**Comment by rfprod** [comment]
@mandarini review please. Please let me know if I modified the correct file, and if I should make any other changes.

**Comment by mandarini** [comment]
Thank you for this PR @rfprod !

File: packages/storybook/src/generators/configuration/project-files/.storybook/main.js__tmpl__ [hunk 1]
```diff
@@ -17,13 +17,35 @@ module.exports = {
...
+       * The issue references:
+       * - https://github.com/storybookjs/storybook/issues/16977#issuecomment-1003399336
+       * - https://github.com/storybookjs/storybook/issues/16977#issuecomment-1004180729
+       * Here's what should be removed:
+       * {
+       *   test: /\.html$/,
+       *   loader: './node_modules/raw-loader/dist/cjs.js',
+       *   exclude: /\.async\.html$/
+       * },
+       */
```
Comment (mandarini) [inline packages/storybook/src/generators/configuration/project-files/.storybook/main.js__tmpl__:39]: Do you think we could make this comment smaller, or maybe remove the github issue references/links, and just keep the explanation? (the Jit breaking, and what needs to be removed for example)?

Comment (rfprod) [reply packages/storybook/src/generators/configuration/project-files/.storybook/main.js__tmpl__:39]: Sure, I will remove it in a while once I'm around my laptop.

Comment (mandarini) [reply packages/storybook/src/generators/configuration/project-files/.storybook/main.js__tmpl__:39]: Sure, no rush! I'll take a look tomorrow my morning in any case! :D

Comment (rfprod) [reply packages/storybook/src/generators/configuration/project-files/.storybook/main.js__tmpl__]: @mandarini I made the change you requested.

**Comment by mandarini** [comment]
Works like a charm, thank you!

**Pull request merged**